napari/layers/_shapes_layer/model.py

```diff
@@ -183,7 +183,9 @@ def __init__(self, data, *, shape_type='rectangle', edge_width=1,
         # Freeze refreshes to prevent drawing before the layer is constructed
         with self.freeze_refresh():
             # Add the shape data
-            self.data = ShapeList()
+            self.nd_data = {'2D': ShapeList()}
+            self.nd_data_dim = 2
+            self._data = self.nd_data['2D']
```

Comment:
As mentioned above, I don't think we should default to 2D and then modify — the dimensionality should be in the constructor.

Reply:
done